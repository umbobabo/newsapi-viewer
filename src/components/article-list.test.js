import React from "react";
import ArticlesList from "./article-list";
import renderer from "react-test-renderer";

const article = {
  source: {
    id: "the-telegraph",
    name: "The Telegraph"
  },
  author: "Jhonny Deep",
  title: "test test test test",
  description: "A description",
  url: "https://www.telegraph.co.uk/news/2018/04/25/test-test-test-test/",
  urlToImage: "http://anUrl",
  publishedAt: "2018-04-25T16:07:20Z"
};

const articles = [article, article, article];

describe("ArticlesList", () => {
  it("renders a list of articles", () => {
    const tree = renderer.create(<ArticlesList articles={articles} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
