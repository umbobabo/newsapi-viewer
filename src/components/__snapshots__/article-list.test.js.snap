// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlesList renders a list of articles 1`] = `
<ul>
  <li>
    <div>
      <div
        className="MuiPaper-root-4 MuiPaper-elevation2-8 MuiPaper-rounded-5 MuiCard-root-3 SimpleMediaCard-card-1"
      >
        <div
          className="MuiCardMedia-root-31 SimpleMediaCard-media-2"
          src={undefined}
          style={
            Object {
              "backgroundImage": "url(\\"http://anUrl\\")",
            }
          }
          title="test test test test"
        />
        <div
          className="MuiCardContent-root-33"
        >
          <h2
            className="MuiTypography-root-34 MuiTypography-headline-39 MuiTypography-gutterBottom-51"
          >
            test test test test
          </h2>
          <p
            className="MuiTypography-root-34 MuiTypography-body1-43"
          >
            2018-4-25 17:07:20
          </p>
          <p
            className="MuiTypography-root-34 MuiTypography-body1-43"
          >
            A description
          </p>
        </div>
        <div
          className="MuiCardActions-root-58"
        >
          <a
            className="MuiButtonBase-root-76 MuiButton-root-60 MuiButton-flatPrimary-62 MuiButton-sizeSmall-73 MuiCardActions-action-59"
            href="https://www.telegraph.co.uk/news/2018/04/25/test-test-test-test/"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
          >
            <span
              className="MuiButton-label-61"
            >
              Learn More
            </span>
            <span
              className="MuiTouchRipple-root-79"
            />
          </a>
        </div>
      </div>
    </div>
  </li>
  <li>
    <div>
      <div
        className="MuiPaper-root-4 MuiPaper-elevation2-8 MuiPaper-rounded-5 MuiCard-root-3 SimpleMediaCard-card-1"
      >
        <div
          className="MuiCardMedia-root-31 SimpleMediaCard-media-2"
          src={undefined}
          style={
            Object {
              "backgroundImage": "url(\\"http://anUrl\\")",
            }
          }
          title="test test test test"
        />
        <div
          className="MuiCardContent-root-33"
        >
          <h2
            className="MuiTypography-root-34 MuiTypography-headline-39 MuiTypography-gutterBottom-51"
          >
            test test test test
          </h2>
          <p
            className="MuiTypography-root-34 MuiTypography-body1-43"
          >
            2018-4-25 17:07:20
          </p>
          <p
            className="MuiTypography-root-34 MuiTypography-body1-43"
          >
            A description
          </p>
        </div>
        <div
          className="MuiCardActions-root-58"
        >
          <a
            className="MuiButtonBase-root-76 MuiButton-root-60 MuiButton-flatPrimary-62 MuiButton-sizeSmall-73 MuiCardActions-action-59"
            href="https://www.telegraph.co.uk/news/2018/04/25/test-test-test-test/"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
          >
            <span
              className="MuiButton-label-61"
            >
              Learn More
            </span>
            <span
              className="MuiTouchRipple-root-79"
            />
          </a>
        </div>
      </div>
    </div>
  </li>
  <li>
    <div>
      <div
        className="MuiPaper-root-4 MuiPaper-elevation2-8 MuiPaper-rounded-5 MuiCard-root-3 SimpleMediaCard-card-1"
      >
        <div
          className="MuiCardMedia-root-31 SimpleMediaCard-media-2"
          src={undefined}
          style={
            Object {
              "backgroundImage": "url(\\"http://anUrl\\")",
            }
          }
          title="test test test test"
        />
        <div
          className="MuiCardContent-root-33"
        >
          <h2
            className="MuiTypography-root-34 MuiTypography-headline-39 MuiTypography-gutterBottom-51"
          >
            test test test test
          </h2>
          <p
            className="MuiTypography-root-34 MuiTypography-body1-43"
          >
            2018-4-25 17:07:20
          </p>
          <p
            className="MuiTypography-root-34 MuiTypography-body1-43"
          >
            A description
          </p>
        </div>
        <div
          className="MuiCardActions-root-58"
        >
          <a
            className="MuiButtonBase-root-76 MuiButton-root-60 MuiButton-flatPrimary-62 MuiButton-sizeSmall-73 MuiCardActions-action-59"
            href="https://www.telegraph.co.uk/news/2018/04/25/test-test-test-test/"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
          >
            <span
              className="MuiButton-label-61"
            >
              Learn More
            </span>
            <span
              className="MuiTouchRipple-root-79"
            />
          </a>
        </div>
      </div>
    </div>
  </li>
</ul>
`;
